scripts/check-license-header.sh

```diff
@@ -36,13 +36,15 @@ expected_file_header_template="@@===--------------------------------------------
 
 paths_with_missing_license=( )
 
-file_paths=$(git ls-files $(cat .licenseignore | xargs -I% printf ":(exclude)% "))
+excludes=$(xargs -I% printf ":(exclude)% " < .licenseignore)
```

Comment:
this won't be correct with files with spaces, quotes etc

Reply:
The previous one also wasn't correct then right?

Reply:
indeed, neither of them are correct

Reply:
To make it correct, you'd need something like

```
declare -a excludes=()
while IFS=$'\n' read -r -a line; do excludes+=( ":(exclude)$(printf "%q" "$line")" ); done

[...]

git ls-files "${excludes[@]}"
```

untested, but should be roughly right

Reply:
This is working when filenames have spaces in the include file.

```
eval git ls-files $(cat IGNOREFILE | tr '\n' '\0' | xargs -0 -I% printf '":(exclude)%" ')
```

Reply:
```
❯ cat IGNOREFILE | tr '\n' '\0' | xargs -0 -I% printf '":(exclude)%" ' | xargs git ls-files
```
no eval

Reply:
yeah, that one breaks on files with `"`s in it though. Not implying that it's super important but it's also not correct.

e.g.

```
$ cat ignorefile 
test.sh
foo " bar
foo bar
foobar
'
test.sh
johannes:/tmp/foo (main)
$ cat ignorefile | tr '\n' '\0' | xargs -0 -I% printf '":(exclude)%" ' | xargs git ls-files
xargs: unterminated quote
```

Reply:
At this point I consider this script good enough :D I am done spending my time handling all cases. We control the file names so we can just not use `"` in a filename

Reply:
Sure, I did however suggest something that might actually work. But your call